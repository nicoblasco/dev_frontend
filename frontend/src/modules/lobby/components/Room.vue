<template>
    <div>
        <div class="btn-group-vertical">
            <span>Click to change member ðŸ‘‡</span>
            <div v-for="m in members" :key="m">
                <button type="button" 
                    :class="{'btn btn-success' : m.name == currentUser,  'btn btn-outline-secondary' : m.name != currentUser}"
                    :disabled ="isAbleToVote(m)"
                    @click="joinPlayer(m.name)"
                    > 
                    {{m.name}}  {{completeValue(m)}}
                </button>                
            </div>
        </div>
    </div>
</template>

<script>
export default {
    name: 'Room',
    props:['members','currentUser','join'] ,    
    methods: {
        completeValue(member) {
          return member.value && member.name == this.currentUser  ? '-'+ member.value  : null
        },
        async joinPlayer(member){
          this.join(member)
        },
        isAbleToVote(member){
          return  member.name == this.currentUser || (member.status =='voted' || member.status=='passed')
        }        
    },   
}
</script>

<style scoped>
 button {
    width: 200px;
    margin-bottom: 2px;
 }
</style>