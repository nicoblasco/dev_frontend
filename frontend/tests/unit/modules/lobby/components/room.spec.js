import {shallowMount} from '@vue/test-utils'
import Room from '@/modules/lobby/components/Room'

describe('Unit test en Room', () => {
    
    test('debe de renderizar el componente correctamente ', () => {

        const wrapper =  shallowMount(Room)

        expect(wrapper.html()).toMatchSnapshot()
    })
})