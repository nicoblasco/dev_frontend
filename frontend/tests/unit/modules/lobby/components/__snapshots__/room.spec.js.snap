// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test en Room debe de renderizar el componente correctamente  1`] = `
<div>
  <div class="btn-group-vertical"><span>Click to change member ðŸ‘‡</span></div>
</div>
`;
